dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java

```diff
@@ -103,6 +118,111 @@
         return iterate(query);
     }
 
+    enum OP {equals,not_equals,like,not_like,contains,doesnt_contain,exists,doesnt_exist,matches,doesnt_match;}
+    
+    @Override
+    public Iterator<Item> findByMetadataQuery(Context context, List<String> query_field, List<String> query_op, List<String> query_val, List<String> collection_ids, String regexClause) throws SQLException {
```

Comment:
Kevin, thanks for the feedback.  It has been interesting to convert from the prior version of the code that packaged up a long SQL string with many subqueries.

I agree that it makes sense to change the signature of collection_ids to List<UUID>.

I will do some brainstorming on the MetadataField piece.  I would like to support 3 types of queries:

* schema.element.qualifier (qualifier specified)
* schema.element (null qualifier)
* schema.element.* (wildcard qualifier)
* (any field)

Would you suggest List<List<MetadataField>> as the signature for query_field?  (An empty list would signify all fields).